.circleci/tests/test_despike.py

```diff
@@ -17,3 +25,93 @@ def test_data_availability(data_dir, working_dir, output_dir):
     boldfile = data_dir + "/withoutfreesurfer/sub-01/func/" \
         "sub-01_task-mixedgamblestask_run-1_space-MNI152NLin2009cAsym_desc-preproc_bold.nii.gz"
     assert op.exists(boldfile)
+
+
+def test_nifti_despike(data_dir):
+    """Test Nifti despiking."""
+    # Read in the necessary inputs
+    boldfile = data_dir + "/withoutfreesurfer/sub-01/func/" \
+        "sub-01_task-mixedgamblestask_run-1_space-MNI152NLin2009cAsym_desc-preproc_bold.nii.gz"
+    maskfile = data_dir + "/withoutfreesurfer/sub-01/func/" \
+        "sub-01_task-mixedgamblestask_run-1_space-MNI152NLin2009cAsym_desc-brain_mask.nii.gz"
+    # Create some spikes in the second voxel
+    file_data = read_ndata(boldfile, maskfile)
+    voxel_data = file_data[2, :]  # a voxel across time
+    voxel_data_mean = np.mean(voxel_data)
+    voxel_data_std = np.std(voxel_data)
+    voxel_data[2] = voxel_data_mean + 10 * voxel_data_std
+    voxel_data[3] = voxel_data_mean - 10 * voxel_data_std
+    # What's the min and max, i.e: the amplitude of the spikes created?
+    spiked_min = min(voxel_data)
+    spiked_max = max(voxel_data)
+    # Let's write this temp file out for despiking
+    file_data[2, :] = voxel_data
+    tempdir = tempfile.mkdtemp()
+    filename = tempdir + "/spikedfile.nii.gz"
+    write_ndata(data_matrix=file_data, mask=maskfile, template=boldfile, TR=0.8, filename=filename)
+    spikedfile = filename
+    # Let's despike the image and write it out to a temp file
+    tempdir = tempfile.mkdtemp()
+    os.system("3dDespike -NEW -prefix " + tempdir + "/3dDespike.nii.gz " + spikedfile)
```

Comment:
It would be better to use the interface here, I think. We may use different settings in the interface that could be missed in a direct call.

Reply:
There was a problem getting the interface to work within the pytests and save out the files - but this command was copy-pasted from what the interface runs!